django_project/changes/tests/test_views.py

```diff
@@ -752,8 +752,368 @@ def test_VersionDelete_no_login(self):
+    def tearDown(self):
+        """
+        Teardown after each test.
+        """
+        self.project.delete()
+        self.version.delete()
+        self.category.delete()
+        self.user.delete()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionListView_with_normal_user(self):
+        """
+        Test if a normal user can view a list of version entries.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-list', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        expected_templates = [
+            'version/list.html', u'changes/version_list.html'
+        ]
+
+        self.assertEqual(
+            response.status_code, 200)
+        self.assertEqual(
+            response.template_name, expected_templates)
+        self.assertEqual(
+            response.context_data['object_list'][0],
+                         self.version)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionCreateView_with_normal_user(self):
+        """
+        Test if a normal user can create a list of version entries.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        self.assertEqual(
+            response.status_code, 200)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionUpdateView_with_normal_user(self):
+        """
+        Test if normal user can update a version entry.
+        """
+        self.client.login(username='sonlinux', password='password')
+
+        # lets ensure the user updating owns the project.
+        if self.client == ProjectF.owner:
+            response = self.client.get(reverse('category-delete', kwargs={
+                'slug': self.category.slug,
+                'project_slug': self.category.project.slug
+            }))
+
+            self.assertEqual(
+                response.status_code, 200)
+            expected_templates = [
+                'category/delete.html'
+            ]
+
+            self.assertEqual(
+                response.template_name, expected_templates)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionDeleteView_with_normal_user(self):
+        """
+        Test if a normal user can delete a version entry.
+        """
+        self.client.login(username='sonlinux', password='password')
+
+        response = self.client.get(reverse('category-delete', kwargs={
+            'slug': self.category.slug,
+            'project_slug': self.category.project.slug
+        }))
+
+        self.assertEqual(
+            response.status_code, 200)
+        expected_templates = [
+            'category/delete.html'
+        ]
+
+        self.assertEqual(
+            response.template_name, expected_templates)
+
```

Comment:
You should have two tests I think.
One where the project belongs to user 1, and user 1 try to remove it: OK
One where the project belongs to user2, and user 1 try to remove it: FORBIDDEN

Reply:
Thanks @Gustry 
Updating